video/out/cocoa-cb/window.swift

```diff
@@ -120,6 +120,13 @@ class Window: NSWindow, NSWindowDelegate {
 
         setTitleBarStyle(Int(mpv.macOpts!.macos_title_bar_style))
         contentView!.addSubview(titleBarEffect!, positioned: .above, relativeTo: nil)
+
+        if #available(OSX 10.14, *) {
+            // Handle run-time changes of the system theme (Dark Mode / Light Mode)
+            DistributedNotificationCenter.default().addObserver( forName: NSNotification.Name("AppleInterfaceThemeChangedNotification"), object: nil, queue: nil) { notification in
+                    self.setTitleBarStyle(Int(self.mpv.macOpts!.macos_title_bar_style))
```

Comment:
that won't work. that options struct (`macOpts`) is only usable on init since it only represents the options at that time. runtime changes do not change that value.

DistributedNotificationCenter seems to be available before 10.14 even. so there shouldn't be a need to check for it? a quick search also didn't give me any official resources on `AppleInterfaceThemeChangedNotification` and only [these](https://developer.apple.com/documentation/foundation/nsnotification/name) notifications names are supposedly available? Apple recommends to use a key value observer here, per the docs you linked.